griptape/engines/eval/eval_engine.py

```diff
@@ -0,0 +1,144 @@
+        default=Factory(lambda: [EvalEngine.Param.INPUT, EvalEngine.Param.ACTUAL_OUTPUT])
+    )
+    generate_steps_system_template: J2 = field(default=Factory(lambda: J2("engines/eval/steps/system.j2")))
+    generate_steps_user_template: J2 = field(default=Factory(lambda: J2("engines/eval/steps/user.j2")))
+    generate_results_system_template: J2 = field(default=Factory(lambda: J2("engines/eval/results/system.j2")))
+    generate_results_user_template: J2 = field(default=Factory(lambda: J2("engines/eval/results/user.j2")))
+
+    @criteria.validator  # pyright: ignore[reportAttributeAccessIssue, reportOptionalMemberAccess]
+    def validate_criteria(self, _: Attribute, value: Optional[str]) -> None:
+        if value is None:
+            if self.evaluation_steps is None:
+                raise ValueError("either criteria or evaluation_steps must be specified")
+            return
+
+        if self.evaluation_steps is not None:
+            raise ValueError("can't have both criteria and evaluation_steps specified")
+
+        if not value:
+            raise ValueError("criteria must not be empty")
+
+    @evaluation_steps.validator  # pyright: ignore[reportAttributeAccessIssue, reportOptionalMemberAccess]
+    def validate_evaluation_steps(self, _: Attribute, value: Optional[list[str]]) -> None:
+        if value is None:
+            if self.criteria is None:
+                raise ValueError("either evaluation_steps or criteria must be specified")
+            return
+
+        if self.criteria is not None:
+            raise ValueError("can't have both evaluation_steps and criteria specified")
+
+        if not value:
+            raise ValueError("evaluation_steps must not be empty")
+
+    @evaluation_params.validator  # pyright: ignore[reportAttributeAccessIssue]
+    def validate_evaluation_params(self, attribute: Attribute, value: list[EvalEngine.Param]) -> None:
+        if not value:
+            raise ValueError("evaluation_params must not be empty")
+
+        if EvalEngine.Param.INPUT not in value:
+            raise ValueError("Input is required in evaluation_params")
+
+        if EvalEngine.Param.ACTUAL_OUTPUT not in value:
+            raise ValueError("Actual Output is required in evaluation_params")
+
+    def __attrs_post_init__(self) -> None:
+        if self.evaluation_steps is None:
+            with validators.disabled():
+                self.evaluation_steps = self._generate_steps()
+
+    def evaluate(self, **kwargs) -> tuple[float, str]:
+        if not all(key.upper() in EvalEngine.Param.__members__ for key in kwargs):
+            raise ValueError("All keys in kwargs must be a member of EvalEngine.Param")
+
+        if EvalEngine.Param.INPUT.name.lower() not in kwargs:
+            raise ValueError("Input is required for evaluation")
+        if EvalEngine.Param.ACTUAL_OUTPUT.name.lower() not in kwargs:
+            raise ValueError("Actual Output is required for evaluation")
```

Comment:
agreed with @dylanholmes, seems like we're validating the kwargs here anyway, why the extra enum for the parameters?